adafruit_st7735/st7735r.py

```diff
@@ -47,35 +56,38 @@
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_ST7735.git"
 
-
-_INIT_SEQUENCE = (
-    b"\x01\x80\x96" # SWRESET
-    b"\x11\x80\xff" # SLPOUT
+_INIT_SEQUENCE = bytearray(
+    b"\x01\x80\x96" # SWRESET and Delay 150ms
+    b"\x11\x80\xff" # SLPOUT and Delay
     b"\xb1\x03\x01\x2C\x2D" # _FRMCTR1
-    b"\xb2\x03\x01\x2C\x2D" #
-    b"\xb3\x06\x01\x2C\x2D\x01\x2C\x2D"
+    b"\xb2\x03\x01\x2C\x2D" # _FRMCTR2
+    b"\xb3\x06\x01\x2C\x2D\x01\x2C\x2D" # _FRMCTR3
     b"\xb4\x01\x07" # _INVCTR line inversion
     b"\xc0\x03\xa2\x02\x84" # _PWCTR1 GVDD = 4.7V, 1.0uA
     b"\xc1\x01\xc5" # _PWCTR2 VGH=14.7V, VGL=-7.35V
     b"\xc2\x02\x0a\x00" # _PWCTR3 Opamp current small, Boost frequency
     b"\xc3\x02\x8a\x2a"
     b"\xc4\x02\x8a\xee"
     b"\xc5\x01\x0e" # _VMCTR1 VCOMH = 4V, VOML = -1.1V
-    b"\x2a\x00" # _INVOFF
+    b"\x20\x00" # _INVOFF
     b"\x36\x01\x18" # _MADCTL bottom to top refresh
     # 1 clk cycle nonoverlap, 2 cycle gate rise, 3 sycle osc equalie,
     # fix on VTL
     b"\x3a\x01\x05" # COLMOD - 16bit color
     b"\xe0\x10\x02\x1c\x07\x12\x37\x32\x29\x2d\x29\x25\x2B\x39\x00\x01\x03\x10" # _GMCTRP1 Gamma
     b"\xe1\x10\x03\x1d\x07\x06\x2E\x2C\x29\x2D\x2E\x2E\x37\x3F\x00\x00\x02\x10" # _GMCTRN1
-    b"\x2a\x03\x02\x00\x81" # _CASET XSTART = 2, XEND = 129
-    b"\x2b\x03\x02\x00\x81" # _RASET XSTART = 2, XEND = 129
     b"\x13\x80\x0a" # _NORON
     b"\x29\x80\x64" # _DISPON
 )
 
-class ST7735(displayio.Display):
-    """ST7735 driver for ST7735R Green tabs"""
-    # TODO(tannewt): Add support for Red tabs and non-R chips. https://github.com/adafruit/Adafruit-ST7735-Library/blob/master/Adafruit_ST7735.cpp
-    def __init__(self, bus, *, width, height):
-        super().__init__(bus, _INIT_SEQUENCE, width=width, height=height, colstart=2)
+# pylint: disable=too-few-public-methods
+class ST7735R(displayio.Display):
+    """ST7735 driver for ST7735R"""
+    def __init__(self, bus, *, init=None, **kwargs):
+        """
+        :param bytes init: (Optional) An extra init sequence to append (default=None)
```

Comment:
I'm not a fan of this init argument because it leaks the binary blob init stuff. Display already has a rotation kwarg that can be used for rotation. (It's very slow for OnDiskBitmaps atm but we can improve it later.) If we also need a toggle for RGB or BGR then let's add it explicitly rather than generically.

By using Display for rotation we align our display memory accesses with the native refresh direction and make it easier to make updates tear-free in the future.

Advanced users can always use the display bus themselves to send extra commands too.

Reply:
I think the rotation doesn’t work for some displays like the 3.5 FeatherWing. Also, this was written before I dug into the actual CP code and now that I’m more comfortable with that, making some changes shouldn’t be a problem.

Reply:
So far in my testing, only a few displays needed to be initialized as BGR. The only driver that was complicated though was this one. This particular parameter is a little messy since it's also combined with the display rotation. There's 3 ways of doing this that I can see:

1. Add a BGR parameter to the init function in the driver itself that adds an init code with the default rotation but sets BGR if true and make use of the rotation option in the driver.  
2. Make changes to the way the colors are converted if this flag is toggled when they are 565 encoded as BGR instead of RGB. I don't know if colors are always converted, so if they're not this may not work.
3. Attempt to run the command inside of CP with default rotation if BGR is passed. This option could get messy for non-MIPI displays.

Personally my favorite option is #1 because the change is only in the driver. #2 is my second favorite, but I'm not sure if it will always work. #3 seems a bit sketchy and doesn't provide much benefit over #1, and is less flexible. 

Another idea is we could always start with option #1 and implement option #2 after 4.0 stable is released.

Reply:
I like 1 but want to avoid MADCTL for rotation because it only impacts the memory access orientation, not screen refresh. If we always allow displayio to rotate "in software" rather than using MADCTL we can ensure our memory updates align with screen updates. This also means we can know in what direction the hardware scrolling of the display occurs.

Reply:
I agree, which is why I’m only using MADCTL for BGR vs RGB and leaving rotation at default.